### PR TITLE
Add notes on (in development) ActiveMQ 5.17 release

### PR DESCRIPTION
**Pull request by ehossack-aws**
This is not intended to be an exhaustive commit, but just add the core pages required so that we can collate information as it evolves for 5.17.

As the amount of changes increase/decrease we can adjust these pages, and ultimately remove the "(current development branch)" moniker

### CODE REVIEW

File: src/new-features-in-517.md
Comment (jbonofre) [inline src/new-features-in-517.md:13]: I would add:
* Spring 5 update (done)
* Log4j2 update (in progress)

**Comment by mattrpav** [comment]
FYI-- I think you need to rebase your PR so it doesn't include the JMS tracking page commit

File: src/new-features-in-517.md
Comment (mattrpav) [reply src/new-features-in-517.md:13]: * JMS 2.0
* JUnit 5 support

**Comment by ehossack-aws** [comment]
> FYI-- I think you need to rebase your PR so it doesn't include the JMS tracking page commit

I actually intentionally included it here - we can always make further updates to that page later on!

File: src/new-features-in-517.md
Comment (ehossack-aws) [reply src/new-features-in-517.md:13]: Let's focus on what the current state is:
* Spring 5 <- will add 
* Log4j2 <- we can add this once it's merged
* JMS 2.0 <- already on
* JUnit 5 <- I haven't seen this done yet? We can always update this once it's merged

* moving the page <- let's do this all at once in a commit so we don't have a weird heterogenous link hierarchy.

**Comment by michaelandrepearce** [comment]
Im cautious about this. Release docs should be about what has been completed only.

**Comment by mattrpav** [comment]
@michaelandrepearce I think calling it 'new features' maybe misleading. The intent here is "initiatives" or "planned efforts".

**Comment by gemmellr** [comment]
> > FYI-- I think you need to rebase your PR so it doesn't include the JMS tracking page commit
> 
> I actually intentionally included it here - we can always make further updates to that page later on!

A different-but-similar JMS 2 page was already added, in #54. Why would there now be two?

> Im cautious about this. Release docs should be about what has been completed only.

I share Mikes caution. I also find it odd the first JMS2 page added talks to targetting 5.17.0.RC1 (implying something before RC1?) for stuff that wont initially be implemented, while this one has jumped to mentioning 5.18.0+ for the same stuff  (which is still everything, so I find the page misleading).

**Comment by gemmellr** [comment]
I added this comment to #54 after it was merged (when I noticed it), I think it applies equally here:

>JB previously suggested he wanted to move the component specific docs etc into the component area, as part of a general tidyup and simplification of the docs. While the overall effort doesnt seem to have progressed as yet, as a rather component-specific page and probably the first distinct new page I can think of being created in a while, I think it would probably make sense to start the trend with this?

In short, starting to head towards the goal with any new stuff, rather than adding more stuff the old way that then needs moved/reworked later.

**Comment by ehossack-aws** [comment]
> A different-but-similar JMS 2 page was already added, in #54. Why would there now be two?

Ahah! I missed this PR - I had started writing mine a while back, but didn't raise this PR. Will merge/update delete. Thanks for the link robbie

> > Im cautious about this. Release docs should be about what has been completed only.
> 

Both of you make good points. Can you make some suggestions on how to make it clearer than this work is for an unreleased version? My intent with "current development branch" was to make that clear.

I could potentially not have this in the navbar for now.

Really our intent here is to start collating features for release, and have us on top of the documentation for the release.
Another approach could be to maintain a feature branch that we merge in at release time, but this seems like slightly more effort given the ongoing reorganization work. Up for suggestions - could we set up a staging site? That would also be helpful for visual previews e.g. news feed PRs.

> In short, starting to head towards the goal with any new stuff, rather than adding more stuff the old way that then needs moved/reworked later.

Happy to defer to this opinion. We're talking about 1 more `cp` command here, so I can't believe that this is much more rework, but at least we'll start the trend.

**Comment by mattrpav** [comment]
How about we label it 'Work In Progress'?  the pages could be wip-517x, wip-516x, etc, wip-218x (artemis)

**Comment by ehossack-aws** [comment]
> How about we label it 'Work In Progress'? the pages could be wip-517x, wip-516x, etc, wip-218x (artemis)

+1 from me

File: src/components/classic/releases/new-features-in-517.md
Comment (mattrpav) [inline src/components/classic/releases/new-features-in-517.md:12]: Let's label this 'Work in Progress' here.. we can then move items that have merged PR's to the "New" or "Removed" sections below. From a work flow perspective, we can have a convention where WIP is something that has a PR, New/Removed Features is for merged PRs and Planned is JIRA only at the given time period.

Work in Progress:
- item a
- item b

New Features completed:
- item c
- tem d

Removed Features completed:
- item e

Planned features:
- item f
- item g

then the upgrading.. etc below

**Review by mattrpav — CHANGES_REQUESTED** [review]

File: src/components/classic/releases/new-features-in-517.md
Comment (ehossack-aws) [reply src/components/classic/releases/new-features-in-517.md:12]: I don't think anybody really wants to go and update the website anytime they complete a task tbh. It's pretty easy just to look at the jira if you want more detail about what's specifically in progress.

I think this is clear enough as is?

Comment (michaelandrepearce) [reply src/components/classic/releases/new-features-in-517.md:12]: I prefer 10x the word wip than release. Agreeing with Matt here.

Comment (michaelandrepearce) [reply src/components/classic/releases/new-features-in-517.md:12]: I quite like Kafkas approach where they have kips but a kip is never in the release unless its fully merged to main. Also there is nothing on the core site about kip stuff, it only has release information this avoids confusion to end users. Kips are only tracked in seperate dev wiki and jira, as its only dev until its released.

**Comment by michaelandrepearce** [comment]
I am questioning if this at all should be on the main site. For dev we use wiki, jira and github anyone whos wanting to be a dev would be using that anyhow.

We shouldnt really be exposing end users (which main site is for) to development stuff. Its after all why we have user lists and dev lists on mailing.

**Review by michaelandrepearce — CHANGES_REQUESTED** [review]
Marking this so it doesn't get merged like a recent previous website pr did
 whilst outstanding discussions.

**Comment by mattrpav** [comment]
I don't think we want to track all JIRAs or changes here. I think having something to track initiatives and user impacting things is the intent-- think whiteboard. Things like new config flags, dropped config flags, etc. 

This spans several devs, so having a place where each person can update their part makes less work on the part of whomever cuts the release.

With all the JDK, JEE, problematic language, etc.. its a lot to pull from JIRAs and summaries. So having a means to update this as we go would be handy-- imho.

**Comment by michaelandrepearce** [comment]
I really dont think it should be on public site, happy to have it in wiki like kafka team do, but until something is released and voted on i question if it should be really be on core site.

**Comment by ehossack-aws** [comment]
I think I’m in agreement with both of you actually - we definitely want to track/update work here across devs, but @michaelandrepearce , you’re right it’s a bit odd to do this on a public site. 

So let’s go with a feature branch I’d suggest? Pretty much exactly what they’re for, collating changes, iterating on stuff that might change considerably before public consumption. 

then we can focus on just presenting the information as if it were a release, and when we’re ready to ship 5.17.X, we just hit merge and boom, we’re done!

**Comment by ehossack-aws** [comment]
Seems like we're agreement that this isn't the best approach to take (summarizing a discussion on the apache slack:

- we don't want to advertise features that aren't approved by PMC
- Jira reports can fill a 'live' in-progress overview
- we still don't have a good solution for better release docs -> ideally we move to having them in the repo, but until then, there's no great solution, and this is something to think about.
)

**Pull request closed without merging**
